src/MongoDB.Driver.Core/Core/WireProtocol/IBinaryDocumentFieldDecryptor.cs

```diff
@@ -0,0 +1,42 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MongoDB.Driver.Core.WireProtocol
+{
+    /// <summary>
+    /// Interface for decrypting fields in a binary document.
+    /// </summary>
+    public interface IBinaryDocumentFieldDecryptor
```

Comment:
I'm tempted to rename this interface `IBinaryResponseFieldDecryptor` because AFAIK it is only used to decrypt responses from the server.

Thoughts?

Reply:
Upon reflection I think `IBinaryDocumentFieldDecryptor` is the right name because the thing that is decrypted is the *document* (not the response, the document has to be extracted from the response in order to be decrypted).